### ISSUE TITLE
Audio output is always going through earpiece on iPhone.

### ISSUE THREAD

**Issue by JozefLipovsky** [aws/amazon-chime-sdk-ios#374]
**Describe the bug**
I tested this using v0.17.0 and v0.16.6 releases. We are using a technique similar to https://github.com/aws/amazon-chime-sdk-ios/issues/89#issuecomment-675796035 to automatically choose audio device based on priority `BT headphones > Wired headphones > BuildInSpeaker > AudioHandset(earpieace) > Other`. Even though I see `.audioBuiltInSpeaker` in available devices and I am specifically selecting it session audio output is always going through the AudioHandset (earpiece) on iPhone.

**To Reproduce**
1. Start a chime session
2. Join the same session with iPad and iPhone
3. On the iPhone observe how the session audio output is always going through the AudioHandset (earpiece)

**Expected behavior**
A session audio output should be going through build in speaker on iPhone when `.audioBuiltInSpeaker` is selected.

**Additional context**

Here is our current setup. We have a ViewModel where we initialize and start session (with call kit disabled). 

Immediately after the session is started we are selecting default audio device for the first time based on `currentSession?.audioVideo.listAudioDevices()` in the `func audioSessionDidStart(reconnecting:)` callback.

Additionally, we are re-setting audio device every time when we receive  `func audioDeviceDidChange(freshAudioDeviceList:)` callback, in this case we are using provided `freshAudioDeviceList` array to select from.

```
final class AudioVideoViewModel {
    // snip...
    private(set) var currentSession: DefaultMeetingSession?
    // snip...


    func startSession(with audioVideoConfiguration: AudioVideoSessionConfiguration) {
        do {
            try currentSession?.audioVideo.start(callKitEnabled: false)
            if !audioVideoConfiguration.controlState.microphoneEnabled {
                mute()
            }

        } catch {
            // snip...
        }
    }

    func selectAudioDevice(availableDevices: [MediaDevice]) {
        guard let session = currentSession else {
            return
        }

        let sortedDevices = availableDevices.sorted(by: { $0.type.rawValue < $1.type.rawValue })
        if let preferredDevice = sortedDevices.first {
            session.audioVideo.chooseAudioDevice(mediaDevice: preferredDevice)
            print("####### Available devices: \(sortedDevices) - selecting: \(preferredDevice)")
        }
    }
}

extension AudioVideoViewModel: AudioVideoObserver {
    // snip...
    func audioSessionDidStart(reconnecting: Bool) {
        let availableDevices = currentSession?.audioVideo.listAudioDevices() ?? []
        selectAudioDevice(availableDevices: availableDevices)
    }

    // snip...
}

extension AudioVideoViewModel: DeviceChangeObserver {
    func audioDeviceDidChange(freshAudioDeviceList: [MediaDevice]) {
        selectAudioDevice(availableDevices: freshAudioDeviceList)
    }
}
```

I checked the SDK implementation of `MediaDevice.swift` and `MediaDeviceType.swift`. To me it looks like `.audioBuiltInSpeaker` and `.audioHandset` device types are configured correctly based on the `AVAudioSessionPortDescription`.  (I can also confirm that v0.17.0 fixed AirPods issue https://github.com/aws/amazon-chime-sdk-ios/issues/344 👍 )

```
    public init(label: String, port: AVAudioSessionPortDescription? = nil) {
        self.label = label
        self.port = port
        if let port = port {
            switch port.portType {
            case .bluetoothLE, .bluetoothHFP, .bluetoothA2DP:
                type = .audioBluetooth
            case .builtInReceiver, .builtInMic: 
                type = .audioHandset            <<<<<<<<<<<<<----------------
            case .headphones, .headsetMic:
                type = .audioWiredHeadset
            case .builtInSpeaker:
                type = .audioBuiltInSpeaker    <<<<<<<<<<<<<----------------
            default:
                type = .other
            }
        } else {
            self.type = .other
        }
    }
```

Andas you can see below in our logs I can see devices exactly as I would expect in each use case, but even though I am using `.audioBuiltInSpeaker` session audio output is always going through the earpiece 

**Logs**

iPad
```
1. Joined session with AirPods connected, audio going through AirPods
####### Available devices: [JL_AirPods - audioBluetooth, Built-in Speaker - audioBuiltInSpeaker, iPad Microphone - audioHandset] - selecting: JL_AirPods - audioBluetooth

2. Disconnected AirPods while in the session,  audio going through iPad speakers
####### Available devices: [Built-in Speaker - audioBuiltInSpeaker, iPad Microphone - audioHandset] - selecting: Built-in Speaker - audioBuiltInSpeaker
```

iPhone
```
1. Joined session with AirPods connected, audio going through AirPods
####### Available devices: [JL_AirPods - audioBluetooth, Built-in Speaker - audioBuiltInSpeaker, iPhone Microphone - audioHandset] - selecting: JL_AirPods - audioBluetooth

2. Disconnected AirPods while in the session,  selecting audioBuiltInSpeaker but audio going through audioHandset / earpiece
####### Available devices: [Built-in Speaker - audioBuiltInSpeaker, iPhone Microphone - audioHandset] - selecting: Built-in Speaker - audioBuiltInSpeaker

3. Connecting AirPods again, audio going through AirPods
####### Available devices: [JL_AirPods - audioBluetooth, Built-in Speaker - audioBuiltInSpeaker, iPhone Microphone - audioHandset] - selecting: JL_AirPods - audioBluetooth
```

**Screenshots**

**Test environment Info (please complete the following information):**
 - Device: iPad Pro (iOS 14.8.1), iPad mini (iOS 15.1), iPhone 6s (iOS 15.1),  iPhone 12 Pro (iOS 15.1)
 - Version AmazonChimeSDK:  0.17.0 / Media 0.130. (I also tested 0.16.6 / Media 0.12.1)
 - Can you reproduce this in the demo app? Yes
If you are reporting a crash:
 - Are you using bitcode supported binaries? NO
 - Please provide full crash logs in the form of a ".crash" or ".ips" file with a binary images section.

**Comment by dylonChime** [comment]
Hi @JozefLipovsky 

I copied your code into our demo and tested choosing the built-in speaker after plugging in some wired earbuds. It seemed to do the right thing. Built-in speaker and mic were used. I don't have a bluetooth device to test with right now, so there could be something special Apple does under the hood for bluetooth devices, like a delayed switching or something. Can you try putting a timer in there or changing the device in a different location? Also, can you try to reproduce with our demo app?

**Comment by JozefLipovsky** [comment]
@dylonChime I tried to use modify `DeviceSelectionModel.swift` to simulate the same auto selection in the demo app and I wasn't able to fully reproduce this.
```
    var selectedAudioDevice: MediaDevice {
        let sortedDevices = audioDevices.sorted(by: { $0.type.rawValue < $1.type.rawValue })
        if let preferredDevice = sortedDevices.first {
            return preferredDevice
        }

        return audioDevices.first!
    }
```

But I found out what was the issue. When we were configuring `AVAudioSession` before starting a Chime session we were only setting category:
```
try AVAudioSession.sharedInstance().setCategory(.playAndRecord, options: .allowBluetooth)
```
I followed the example from demo and now we are also setting mode:
```
try AVAudioSession.sharedInstance().setMode(.videoChat)
```

After this when NO headphones are connected audio output is coming from both built in speaker and earpiece at the same time on a phone. (not like before when it was only coming from the earpiece, and was super quiet)

I am closing this issue now 👍 

But the `.setMode()` setup is probably something that's worth documenting in the [getting-started](https://github.com/aws/amazon-chime-sdk-ios/blob/master/guides/getting_started.md#getting-started), could probably save some time to other users in the future.
